src/Illuminate/Support/Collection.php

```diff
@@ -522,11 +522,16 @@ public function pop()
      * Push an item onto the beginning of the collection.
      *
      * @param  mixed  $value
+     * @param  mixed  $key
      * @return $this
      */
-    public function prepend($value)
+    public function prepend($value, $key = null)
     {
-        array_unshift($this->items, $value);
+        if (is_null($key)) {
+            array_unshift($this->items, $value);
+        } else {
+            $this->items = [$key => $value] + $this->items;
```

Comment:
I would just go on using `array_merge` as this is the syntax we use in `\Illuminate\Support\Collection`